src/openforms/submissions/logic/rules.py

```diff
@@ -161,3 +161,22 @@ def iter_evaluate_rules(
                 else:
                     operation = compile_action_operation(action)
                     yield operation
+
+
+def get_targeted_components(
+    rule: FormLogic,
+    components_map: dict,
+    all_variables: Dict[str, FormVariable],
+    initial_data: dict,
+) -> List[dict]:
+    targeted_components = []
+    for action in rule.actions:
+        action = ActionOperation.from_action(action)
```

Comment:
`compile_action_operation` is used in other places as the public API

```suggestion
        action = compile_action_operation(action)
```

I'd also use a different variable name, since this changes the type of `action` from `dict` to `ActionOperation`, which I expect `mypy`/editors to complain about.